src/services/tradeApiClient.js

```diff
@@ -336,32 +345,44 @@ class TradeApiClient {
    * Process API HTTP request.
    *
    * @param {string} endpointPath API endpoint path and action.
-   * @param {Object} payload Request payload parameters object.
+   * @param {*} payload Request payload parameters object.
    * @param {string} [method] Request HTTP method, currently used only for cache purposes.
    * @param {string} [token] Optional authentication token.
    * @returns {Promise<*>} Promise that resolves Trade API request response.
    *
    * @memberof TradeApiClient
    */
-  async doRequest(endpointPath, payload, method = "", token) {
-    const requestUrl = this.baseUrl + endpointPath;
+  async doRequest(endpointPath, payload, method = "POST", token) {
+    const baseUrl =
+      payload.apiVersion && payload.apiVersion === 2 ? this.baseUrlv2 : this.baseUrlv1;
```

Comment:
Move apiVersion to a separate method param

Reply:
done but this produces a warning fro number of arguments thats why i didn't do it.